### PR TITLE
Fix #3962 - Profiler uses DatabaseSchema & Sample Data fix

### LINKED ISSUE #3962: Schema Name - Adapt Profiler converter

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes :

Fix #3962 

Also fixes:
- invalid options casting in engine creation
- looker config in tests
- sample data airlfow
- Fix https://github.com/open-metadata/OpenMetadata/issues/4062

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/utils/engines.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def get_engine(workflow_source: WorkflowSource, verbose: bool = False) -> Engine
         options = {}
     engine = create_engine(
         get_connection_url(service_connection_config),
-        **options.dict(),
+        **options,
```
Comment (pmbrull) [inline ingestion/src/metadata/utils/engines.py:41]: @ayush-shah @OnkarVO7, in this [PR](https://github.com/open-metadata/OpenMetadata/pull/4027/files#diff-7ba89ab2f4820db88ff3d78709d291699775198b5e7d8ca751144469504cd41fR41) we were adding this `dict()` to `options`. This breaks other ingestions such as with MSSQL.

We have this issue https://github.com/open-metadata/OpenMetadata/issues/4032 to make the engine creation a bit smarter. For now, I am reverting this change, but we should fix the Bigquery ingestion and check how we are passing the parameters.

Comment (pmbrull) [reply ingestion/src/metadata/utils/engines.py:41]: Here I'd do two things to get this right and easy for us and users:
1. We know the parameters we need to prepare the BQ connection, let's move them to the JSON Schema
2. Prepare a singledispatch for the engine creation as we do with the URL. One default process where we unpack the `options`, and specific configs for those services where we have required params, such as BQ or the `http_path` in databricks
WDYT?

Comment (OnkarVO7) [reply ingestion/src/metadata/utils/engines.py:41]: Yes i think both points are ok we can do that

**Comment by pmbrull** [comment]
Fixing sample data ingestion

**Review by ayush-shah — APPROVED** [review]

**Pull request merged**